skops/io/_persist.py

```diff
@@ -33,12 +34,19 @@ def _save(obj: Any, compression: int, compresslevel: int | None) -> io.BytesIO:
         buffer, "w", compression=compression, compresslevel=compresslevel
     ) as zip_file:
         save_context = SaveContext(zip_file=zip_file)
-        state = get_state(obj, save_context)
-        save_context.clear_memo()
-
-        state["protocol"] = save_context.protocol
-        state["_skops_version"] = skops.__version__
-        zip_file.writestr("schema.json", json.dumps(state, indent=2))
+        # If obj is scikeras model save the scikeras model via scikeras
+        if "scikeras" in obj.__class__.__module__:
+            with tempfile.NamedTemporaryFile(mode="w+", newline="") as temp_file:
+                file_name = temp_file.name + ".keras"
+                obj.model.save(file_name)
+                zip_file.write(file_name, "model.keras")
```

Comment:
Oh I see what you're doing here. This unfortunately doesn't work when you put a scikeras model inside a `Pipeline` for instance. This code should go in the same place similar to where we handle `sklearn` and `numpy` specific objects, and be added to our dispatch mechanism somehow. And the created dump files go in the same zipfile which includes the other objects.